src/ledger/LedgerTxn.cpp

```diff
@@ -240,10 +240,12 @@ LedgerKeyMeter::maxReadQuotaForKey(LedgerKey const& key) const
                       key.type() != CONTRACT_DATA);
         return std::numeric_limits<uint32_t>::max();
     }
-    return std::reduce(iter->second.begin(), iter->second.end(), 0,
-                       [&](uint32_t maxReadQuota, size_t const txn) {
-                           return std::max(maxReadQuota, mTxReadBytes.at(txn));
-                       });
+    uint32_t maxReadQuota = 0;
+    for (const auto txnIndex : iter->second)
+    {
+        maxReadQuota = std::max(maxReadQuota, mTxReadBytes.at(txnIndex));
```

Comment:
Storing indexes like this seems to add unnecessary complexity to the code. You could use shared pointers to update quotas for a key, and get rid of mTxReadBytes completely. Then you can use `max_element` here to compare elements directly.

Reply:
Done.